### PR TITLE
 Avoid N+1 queries in comments controller

### PR DESCRIPTION
**Pull request by coolo**

### CODE REVIEW

File: src/api/app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ class CommentsController < ApplicationController
   before_action :find_obj, only: [:index, :create]
 
   def index
-    @comments = @obj.comments.order(:id)
+    @comments = @obj.comments.preload(:user).order(:id)
```
Comment (Ana06) [inline src/api/app/controllers/comments_controller.rb]: any significative change in performance if you use `eager_load` instead? :thinking:

Comment (bgeuken) [inline src/api/app/controllers/comments_controller.rb]: I don't think you need to enforce the preload. Just add an `include(:user)`. As soon as some chained method call triggers the query, the comments will be included / prefetched.

Comment (coolo) [reply src/api/app/controllers/comments_controller.rb]: the difference between includes and preload is marginal - if even existant in this case.

What we might want to use though is join (or eager_load)

Comment (coolo) [reply src/api/app/controllers/comments_controller.rb]: I can't see any difference between preload and includes. But it seems like includes is more known - so let's use this one

**Comment by coolo** [comment]
And just to be clear: we're not talking about a major win in absolute numbers - but it still saves 30% 

Before:
```
frontend_1  | 16:36:15 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.08]   Rendering comments/index.xml.builder
frontend_1  | 16:36:15 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.08]   Comment Load (1.1ms)  SELECT `comments`.* FROM `comments` WHERE `comments`.`commentable_id` = 249724 AND `comments`.`commentable_type` = 'Project' ORDER BY `comments`.`id` ASC
frontend_1  | 16:36:15 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.08]   User Load (0.2ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1
frontend_1  | 16:36:15 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.09]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:15 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.09]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:15 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.09]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:15 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.09]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:15 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.09]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.09]   User Load (0.3ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 260 LIMIT 1
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.09]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.10]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.10]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.10]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.10]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.10]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.10]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.10]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.10]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.11]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.11]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.11]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.11]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.11]   User Load (0.4ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 53113 LIMIT 1
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.11]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.12]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.12]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.12]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.12]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.13]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.13]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.13]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.13]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.13]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.13]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.13]   User Load (0.4ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 2639 LIMIT 1
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.13]   CACHE User Load (0.0ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 40248 LIMIT 1  [["id", 40248], ["LIMIT", 1]]
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.14]   User Load (0.3ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 55435 LIMIT 1
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.14]   User Load (0.2ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 53382 LIMIT 1
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.14]   Rendered comments/_comments.xml.builder (58.3ms)
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.14]   Rendered comments/index.xml.builder (59.2ms)
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.14] no schema found, skipping validation for <ActionController::Parameters {"format"=>"xml", "controller"=>"comments", "action"=>"index", "project"=>"openSUSE:Factory:Staging:M", "method"=>"GET", "type"=>"response"} permitted: false>
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.14] Validate XML response: #<ActionDispatch::Response:0x00000000079f78b8> took 0ms
frontend_1  | 16:36:16 web.1         | [7c1d39ff-8396-47a7-8aa6-2282e28dcb27] [9:0.14] Completed 200 OK in 132ms (Views: 60.9ms | ActiveRecord: 8.6ms | Backend: 0.0ms)
```

After:

```
frontend_1  | 16:37:25 web.1         | [c5d4fdea-8d0b-433a-9707-eed8c9546325] [9:70.10]   Rendering comments/index.xml.builder
frontend_1  | 16:37:25 web.1         | [c5d4fdea-8d0b-433a-9707-eed8c9546325] [9:70.10]   Comment Load (0.5ms)  SELECT `comments`.* FROM `comments` WHERE `comments`.`commentable_id` = 249724 AND `comments`.`commentable_type` = 'Project' ORDER BY `comments`.`id` ASC
frontend_1  | 16:37:25 web.1         | [c5d4fdea-8d0b-433a-9707-eed8c9546325] [9:70.10]   User Load (0.4ms)  SELECT `users`.* FROM `users` WHERE `users`.`id` IN (40248, 260, 53113, 2639, 55435, 53382)
frontend_1  | 16:37:26 web.1         | [c5d4fdea-8d0b-433a-9707-eed8c9546325] [9:70.12]   Rendered comments/index.xml.builder (20.0ms)
frontend_1  | 16:37:26 web.1         | [c5d4fdea-8d0b-433a-9707-eed8c9546325] [9:70.12] no schema found, skipping validation for <ActionController::Parameters {"format"=>"xml", "controller"=>"comments", "action"=>"index", "project"=>"openSUSE:Factory:Staging:M", "method"=>"GET", "type"=>"response"} permitted: false>
frontend_1  | 16:37:26 web.1         | [c5d4fdea-8d0b-433a-9707-eed8c9546325] [9:70.12] Validate XML response: #<ActionDispatch::Response:0x00007f3998064090> took 0ms
frontend_1  | 16:37:26 web.1         | [c5d4fdea-8d0b-433a-9707-eed8c9546325] [9:70.12] Completed 200 OK in 92ms (Views: 23.9ms | ActiveRecord: 5.9ms | Backend: 0.0ms)
```

**Pull request merged**